bindings/python/mappings.py

```diff
@@ -72,14 +73,35 @@ def instance_routes(
     if isinstance(instances, dlite.Instance):
         instances = [instances]
 
+    # FIXME: Bad workaround for adding missing units to the registry.
+    # Too be fixed ASAP.
+    ureg = quantity(1.0, "m").units._REGISTRY
+    unit_definitions = [
+        ("atom_fraction", "atom_fraction = 1"),
+    ]
+    for uname, udef in unit_definitions:
+        if uname not in ureg:
+            ureg.define(udef)
+
+
     sources = {}
     for inst in instances:
         props = {prop.name: prop for prop in inst.meta["properties"]}
         for key, value in inst.properties.items():
             if props[key].unit:
+
+                try:
+                    q = quantity(1.0, props[key].unit)
+                except TypeError:
+                    raise UnknownUnitError(
+                        f"unknown unit '{props[key].unit}' in datamodel: "
+                        f"{inst.meta.uri}"
+                    )
```

Comment:
Yes. Pint raises consistently TypeError from deep down in its code if the unit is unknown.

A better solution would be that the functions in the mappings module takes a pint.UnitRegistry (or maybe better a class wrapped around it) instead of pint.Quantity class as input. Then we could replace the above try-except with

    if props[key].unit not in ureg:
        raise UnknownUnitError(...)

Reply:
So in which test it this being tested?